File: src/data/courses/dsa/arrays-and-hashing/array-manipulation/set-matrix-zeroes/code/python/set_matrix_zeroes_sol1.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        # O(m+n) space — Record zero rows and columns
        m, n = len(matrix), len(matrix[0])
        rows = set(); cols = set()
        for i in range(m):
            for j in range(n):
                if matrix[i][j] == 0: rows.add(i); cols.add(j)
        for i in range(m):
            for j in range(n):
                if i in rows or j in cols: matrix[i][j] = 0

File: src/data/courses/dsa/arrays-and-hashing/array-manipulation/set-matrix-zeroes/code/cpp/set_matrix_zeroes_sol1.cpp
class Solution {
public:
    // O(m+n) space — Record zero rows and columns
    void setZeroes(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        unordered_set<int> rows, cols;
        for (int i = 0; i < m; i++)
            for (int j = 0; j < n; j++)
                if (matrix[i][j] == 0) { rows.insert(i); cols.insert(j); }
        for (int i = 0; i < m; i++)
            for (int j = 0; j < n; j++)
                if (rows.count(i) || cols.count(j)) matrix[i][j] = 0;
    }
};